src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java

```diff
@@ -536,11 +536,17 @@ private long nextIndexWithNextAvailableCycle0(final int cycle) {
             nextIndex = nextIndexWithinFoundCycle(nextCycle);
         else
             try {
-                final int nextCycle0 = queue.nextCycle(this.cycle, direction);
-                if (nextCycle0 == -1)
-                    return Long.MIN_VALUE;
-
-                nextIndex = nextIndexWithinFoundCycle(nextCycle0);
+                int nextCycle0 = this.cycle;
+                while (true) {
+                    nextCycle0 = queue.nextCycle(nextCycle0, direction);
+                    if (nextCycle0 == -1)
+                        return Long.MIN_VALUE;
+
+                    if (cycle(nextCycle0)) {
+                        nextIndex = nextIndexWithinFoundCycle(nextCycle0);
+                        break;
+                    }
```

Comment:
If the next cycle found (according to the caches) can't be loaded, keep traversing forwards until we find one that can (or we reach the end)